js/components/address.js

```diff
@@ -0,0 +1,298 @@
+        }
+      ]
+    },
+    fieldOptions: {
+      type: Array,
+      default: []
+    },
+    selectedFieldOptions: {
+      type: Object,
+      default: {}
+    },
+    addressSuggestions: {
+      type: Array,
+      default: []
+    },
+    choosenAddress: {
+      type: Object,
+      default: {}
+    },
+    addressComponents: {
+      type: Array,
+      default: []
+    },
+    suggestionSelected: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data: function() {
+    return {
+      lastChosenAutocompleteValue: ''
+    };
+  },
+  created: function() {
+    this.separateFieldsName.forEach((field) => {
+      if (!this.selectedFieldOptions[field.key]) {
+        this.$set(this.selectedFieldOptions, field.key, '');
+      }
+    });
+    this.updateDisabledOptions();
+
+    Fliplet.Hooks.on('beforeFormSubmit', this.onBeforeSubmit);
+  },
+  destroyed: function() {
+    Fliplet.Hooks.off('beforeFormSubmit', this.onBeforeSubmit);
+  },
+  mounted: function() {
+    this.initAutocomplete('', this.countryRestrictions);
+    this.onChange();
+    document.addEventListener('click', this.handleClickOutside);
+
+    this.$emit('_input', this.name, this.value, false, true);
+  },
+  methods: {
+    handleClickOutside: function(event) {
+      const suggestionsList = this.$el.querySelector('.google-autocomplete');
+
+      if (suggestionsList && !suggestionsList.contains(event.target) && this.manualInput) {
+        this.addressSuggestions = [];
+        this.suggestionSelected = false;
+      }
+    },
+    selectSuggestion: async function(option) {
+      this.value = option;
+      this.addressSuggestions = [];
+      this.suggestionSelected = true;
+
+      const data = await this.addressField.getAddressComponents(option.id);
+
+      this.addressComponents = data;
+
+      this.addressField.set(option.label);
+      this.updateValue();
+
+      this.onChange();
+    },
+    extractAddressComponents: function(place) {
+      var addressData = {
+        streetNumber: '',
+        streetName: '',
+        city: '',
+        state: '',
+        country: '',
+        postalCode: ''
+      };
+
+      if (place.length) {
+        place.forEach(component => {
+          var types = component.types;
+
+          if (types.includes('street_number')) {
+            addressData.streetNumber = component.long_name;
+          }
+
+          if (types.includes('route')) {
+            addressData.streetName = component.long_name;
+          }
+
+          if (types.includes('locality')) {
+            addressData.city = component.long_name;
+          }
+
+          if (types.includes('administrative_area_level_1')) {
+            addressData.state = component.long_name;
+          }
+
+          if (types.includes('country')) {
+            addressData.country = component.long_name;
+          }
+
+          if (types.includes('postal_code')) {
+            addressData.postalCode = component.long_name;
+          }
```

Comment:
@armine-fliplet , is that purposely assigned with the same value?

Reply:
@dwfliplet Yes)